### ISSUE TITLE
Prettify the number-errbacks errors

### ISSUE THREAD

**Issue by blerner** [brownplt/pyret-lang#758]
Currently all the number-errbacks are just wrappers around throwMessageException.  They can be better.

**Comment by blerner** [comment]
Remaining todo steps:
* In runtime.js, the `NumberErrbacks` object is used by the js-numbers library to throw actual Pyret exceptions.  Every method there currently has signature String -> thrown exception.
* Revise the signatures as needed, grep for the method uses in js-numbers, and change them to match.
* Add whatever new errors and render-fancy-reason/render-reason methods you need in error.arr to match.

**Comment by schanzer** [comment]
Isn't #503 a superset of this now?

**Comment by schanzer** [comment]
I believe https://github.com/jpolitz/pyret-lang/pull/27 will address this, so tagging it here so we can close this issue when it lands

**Comment by blerner** [comment]
No, that PR is about *testing* the numbers library and making sure it doesn't crash; it doesn't affect which errbacks are thrown. They're related PRs, yes, but that one won't change this one at all.
